pandas/tslib.pyx

```diff
@@ -2294,6 +2294,13 @@ cpdef array_to_datetime(ndarray[object] values, errors='raise',
                         iresult[i] = cast_from_unit(val, 'ns')
                     except:
                         iresult[i] = NPY_NAT
+            elif is_bool_object(val) and not is_coerce:
+                if val == NPY_NAT:
+                    iresult[i] = NPY_NAT
+                elif is_raise:
+                    raise ValueError("boolean is not convertible to datetime")
```

Comment:
you need to ``raise TypeError`` so it kicks out to the object loop

add a test that has something like

```pd.to_datetime(['20130101',True])```